### PR TITLE
Add support for APRS Messages

### PR DESCRIPTION
**Pull request by scd31**
Adds support for APRS messages.

Also did some minor refactoring(renamed `APRS**** to Aprs****, Renamed APRSMessage to AprsPacket to make room for the new struct)

### CODE REVIEW

**Comment by Turbo87** [comment]
can you split the formatting and refactoring changes from the feature implementation? mixing everything in one commit/PR makes this quite hard to review 🙏

File: src/message.rs [hunk 1]
```diff
@@ -1,77 +1,63 @@
...
-    pub via: Vec<Callsign>,
-    pub data: APRSData,
+use AprsError;
+use FromStr;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct AprsMessage {
+    pub addressee: String,
+    pub text: String,
+    pub id: Option<u32>,
```
Comment (scd31) [inline src/message.rs:8]: I'll need to update this as `id` can be alphanumeric

**Comment by Turbo87** [comment]
@scd31 is there a specification for this? do you have a reference/URL/something where I can read up on this?

**Comment by scd31** [comment]
@Turbo87 I'm using this: http://www.aprs.org/doc/APRS101.PDF

Messages start on page 71/81.

**Comment by scd31** [comment]
Are there any plans to merge this? I know it needs the conflicts resolved, which I'm happy to fix. Just wondering if I should fork this instead

**Comment by Turbo87** [comment]
sorry for the long silence. the package is not high on my priority list :-/

**Comment by scd31** [comment]
No worries! Just glad it got merged.

File: src/message.rs (hunk 1 above)
Comment (scd31) [reply src/message.rs:8]: I'm going to open another PR to change the id to a String and add `messaging_supported` to the README. Does that work for you @Turbo87 ?

**Pull request merged**